tests/Microsoft.Identity.Test.Unit/PublicApiTests/TelemetryClientTests.cs

```diff
@@ -140,11 +425,41 @@ public async Task AcquireTokenUnSuccessfulTelemetryTestAsync()
 
                 MsalTelemetryEventDetails eventDetails = _telemetryClient.TestTelemetryEventDetails;
                 Assert.AreEqual(ex.ErrorCode, eventDetails.Properties[TelemetryConstants.ErrorCode]);
+                Assert.AreEqual(ex.Message, eventDetails.Properties[TelemetryConstants.ErrorMessage]);
+                Assert.AreEqual(ex.ErrorCodes.AsSingleString(), eventDetails.Properties[TelemetryConstants.StsErrorCode]);
                 Assert.IsFalse((bool?)eventDetails.Properties[TelemetryConstants.Succeeded]);
+
+                //Test for MsalClientException
+                _harness.HttpManager.AddTokenResponse(TokenResponseType.InvalidClient);
+
+                MsalClientException exClient = await AssertException.TaskThrowsAsync<MsalClientException>(
+                    () => _cca.AcquireTokenForClient(null)
+                    .WithAuthority(TestConstants.AuthorityUtidTenant)
+                    .ExecuteAsync(CancellationToken.None)).ConfigureAwait(false);
+
+                Assert.IsNotNull(exClient);
+                Assert.IsNotNull(exClient.ErrorCode);
+
+                eventDetails = _telemetryClient.TestTelemetryEventDetails;
+                Assert.AreEqual(exClient.ErrorCode, eventDetails.Properties[TelemetryConstants.ErrorCode]);
+                Assert.AreEqual(exClient.Message, eventDetails.Properties[TelemetryConstants.ErrorMessage]);
+                Assert.IsFalse((bool?)eventDetails.Properties[TelemetryConstants.Succeeded]);
+
+                //Test for generic Exception
```

Comment:
Trying to figure out the best way to simulate a generic exception for this test.

Reply:
How about - inject a custom HttpClient (via `WithHttpManager` or `WithHttpClientFactory`). See example here https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/tests/Microsoft.Identity.Test.Unit/PublicApiTests/AccountTests.cs#L59 (here it asserts, you can just throw a SocketNotFoundException or smth)_